content/3_Silver/Sorting_Custom.mdx

```diff
@@ -48,626 +46,449 @@ After sorting, it should look like
+<LanguageSection>
+<CPPSection>
+
+There's multiple ways we can do this.
+The most straightforward method is to use a `pair<int, pair<int, int>>`.
+When comparing these, C++ looks at the first element and only uses
+the second element as a tiebreaker.
 
 ```cpp
-#include <bits/stdc++.h>
-using namespace std;
+#include <algorithm>
+#include <iostream>
+#include <vector>
 
-#define f first
-#define s second
+using namespace std;
 
 int main() {
-	int M = 4;
-	vector<pair<int, pair<int, int>>> v;
-	for (int i = 0; i < M; ++i) {
-		int a, b, w;
-		cin >> a >> b >> w;
-		v.push_back({w, {a, b}});
-	}
-	sort(begin(v), end(v));
-	for (auto e : v) cout << e.s.f << " " << e.s.s << " " << e.f << "\n";
-}
-```
+	constexpr int edge_num = 4;
 
-or a `vector` of `array<int,3>`s or `vector<int>`s:
+	vector<pair<int, pair<int, int>>> edges(edge_num);
+	for (auto &[width, edge] : edges) { cin >> edge.first >> edge.second >> width; }
+	sort(edges.begin(), edges.end());
 
-```cpp
-int main() {
-	int M = 4;
-	vector<array<int, 3>> v;  // or vector<vector<int>>
-	for (int i = 0; i < M; ++i) {
-		int a, b, w;
-		cin >> a >> b >> w;
-		v.push_back({w, a, b});
+	for (const auto &[width, edge] : edges) {
+		printf("(%i, %i): %i\n", edge.first, edge.second, width);
 	}
-	sort(begin(v), end(v));
-	for (auto e : v) cout << e[1] << " " << e[2] << " " << e[0] << "\n";
 }
 ```
 
-In Python, we can use a list of lists.
-
-But in Java, we can't sort an `ArrayList` of `ArrayList`s without writing some
-additional code. What should we do?
-
-- If we only stored the edge weights and sorted them, we would have a sorted
-  list of edge weights, but it would be impossible to tell which weights
-  corresponded to which edges.
-- However, if we create a **class** representing the edges and define a **custom
-  comparator** to sort them by weight, we can sort the edges in ascending order
-  while also keeping track of their endpoints.
-
-## Classes
-
-First, we need to define a **class** that represents what we want to sort. In
-our example we will define a class `Edge` that contains the two endpoints of the
-edge and the weight.
-
-<LanguageSection>
-
-<CPPSection>
-
-### C++
-
-A C++ `struct` is the same as a `class` in C++, but all members are public by
-default.
+If you don't like all the accesses to `.first` and `.second`,
+an alternative is to use an `std::vector` or maybe even an
+[`std::array`](https://en.cppreference.com/w/cpp/container/array):
 
 ```cpp
-#include <bits/stdc++.h>
-using namespace std;
-
-struct Edge {
-	int a, b, w;
-};
+#include <algorithm>
+#include <iostream>
+#include <vector>
 
-/* alternatively,
-class Edge {
-    public:
-        int a,b,w;
-};
-*/
+using namespace std;
 
 int main() {
-	int M = 4;
-	vector<Edge> v;
-	for (int i = 0; i < M; ++i) {
-		int a, b, w;
-		cin >> a >> b >> w;
-		v.push_back({a, b, w});
+	constexpr int edge_num = 4;
+
+	// You can replace vector<int> with array<int, 3> if you like
```

Comment:
making a lot of really small vectors is pretty bad practice, so in general just use std::array if u can

Reply:
wait is this like an actual issue

Reply:
nah not a huge issue, might be worth mentioning though